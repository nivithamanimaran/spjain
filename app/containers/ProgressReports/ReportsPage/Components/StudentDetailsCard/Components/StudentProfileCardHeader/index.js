import React from 'react';
import { makeStyles, Box, Grid } from '@material-ui/core';
import KenAutoComplete from '../../../../../../../components/KenAutoComplete';
import ArrowLeft from '../../../../../../../assets/icons/arrowLeft.svg';
import ArrowRight from '../../../../../../../assets/icons/arrowRight.svg';

const useStyles = makeStyles(theme => ({
  actionIcon: {
    cursor: 'pointer',
  },
  avatar: {
    margin: '0 auto',
  },
  text: {
    textAlign: 'center',
  },
  root: {
    padding: 14,
  },
}));

const StudentProfileCardHeader = props => {
  const { data, setStudentInfo, studentInfo } = props;
  const classes = useStyles();

  const handleArrows = event => {
    let ind;
    data.forEach((el, index) => {
      if (studentInfo.ContactId === el.ContactId) {
        ind = index;
      }
    });
    if (event.target.id === 'left') {
      ind = ind - 1;
    } else {
      ind = ind + 1;
    }
    data.forEach((el, index) => {
      if (index === ind) {
        setStudentInfo(el);
      }
    });
  };

  return (
    <Grid
      container
      direction="row"
      className={classes.root}
      alignItems="center"
      justify="space-evenly"
    >
      <Grid item md={2} xs={2} sm={2} className={classes.text}>
        <img
          src={ArrowLeft}
          id="left"
          className={classes.actionIcon}
          onClick={e => {
            handleArrows(e);
          }}
        />
      </Grid>
      <Grid item md={8} xs={8} sm={8} className={classes.text}>
        <KenAutoComplete
          options={data}
          placeholder="Search..."
          setData={setStudentInfo}
        />
      </Grid>
      <Grid item md={2} xs={2} sm={2} className={classes.text}>
        <img
          src={ArrowRight}
          id="right"
          className={classes.actionIcon}
          onClick={e => {
            handleArrows(e);
          }}
        />
      </Grid>
    </Grid>
  );
};
export default StudentProfileCardHeader;
