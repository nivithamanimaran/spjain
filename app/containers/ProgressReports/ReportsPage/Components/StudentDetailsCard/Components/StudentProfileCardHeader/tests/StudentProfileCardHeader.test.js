import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import StudentProfileCardHeader from '../index';
import ThemeProvider from '../../../../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

describe('Student-Details-Card-Header-Page', () => {
  it('Student Details Card Header renders correctly without props', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <StudentProfileCardHeader />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <StudentProfileCardHeader />
        </Router>
      </ThemeProvider>
    );
    const cardHeader = getByTestId('student-profile-card-header');
    expect(cardHeader).toBeTruthy();
    expect(cardHeader.textContent).toBe('');

    const avatar = getByTestId('student-profile-avatar');
    expect(avatar).toBeTruthy();
    expect(avatar.textContent).toBe('');

    const name = getByTestId('student-profile-name');
    expect(name).toBeTruthy();
    expect(name.textContent).toBe('');
    expect(getByTestId('student-profile-card-header')).toMatchSnapshot();
  });

  it('Student Details Card Header renders correctly with props', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <StudentProfileCardHeader />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <StudentProfileCardHeader primary="Fake Name" avatarImageUrl="" />
        </Router>
      </ThemeProvider>
    );

    const cardHeader = getByTestId('student-profile-card-header');
    expect(cardHeader).toBeTruthy();

    const avatar = getByTestId('student-profile-avatar');
    expect(avatar).toBeTruthy();
    expect(avatar.textContent).toBe('F');

    const name = getByTestId('student-profile-name');
    expect(name).toBeTruthy();
    expect(name.textContent).toBe('Fake Name');
  });
});
