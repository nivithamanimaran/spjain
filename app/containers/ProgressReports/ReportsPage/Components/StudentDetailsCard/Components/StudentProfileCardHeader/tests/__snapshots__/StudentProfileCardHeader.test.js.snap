// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student-Details-Card-Header-Page Student Details Card Header renders correctly without props 1`] = `
<div
  class="MuiBox-root MuiBox-root-8"
  data-testid="student-profile-card-header"
>
  <div
    class="MuiBox-root MuiBox-root-9"
    style="justify-content: center;"
  >
    <div
      class="MuiBox-root MuiBox-root-10"
      data-testid="student-profile-avatar"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-11 makeStyles-text-3"
      data-testid="student-profile-name"
    >
      <div
        class="MuiListItemText-root"
      />
    </div>
  </div>
</div>
`;
